src/app/pages/BuySovPage/components/BuyForm/Dialogs/SlippageDialog.tsx

```diff
@@ -23,92 +21,94 @@ interface ISlippageDialogProps {
-}) => {
+  ...props
+}: Props) => {
   const { t } = useTranslation();
-  const [slippageValue, setSlippageValue] = useState(value);
-  const { minReturn } = useSlippage(amount, slippageValue);
-
-  const cancel = () => {
-    setSlippageValue(value);
-    onClose();
-  };
+  const [value, setValue] = useState(props.value);
+  const { minReturn } = useSlippage(amount, value);
 
-  const confirm = () => {
-    onChange(slippageValue);
-    onClose();
-  };
+  const handleChange = useCallback(
+    (value: number) => {
+      setValue(value);
+      onChange(value);
+    },
+    [onChange],
+  );
 
   return (
     <Dialog
       isCloseButtonShown={false}
       isOpen={isOpen}
-      onClose={() => onClose()}
+      onClose={onClose}
       className={styles.dialog}
+      data-action-id={dataActionId}
     >
       <button
         data-close=""
-        onClick={() => onClose()}
-        data-action-id={`${dataActionId}slippageDialog-button-close`}
+        onClick={onClose}
+        data-action-id="buySov-slippageDialog-button-close"
       >
         <span className="tw-sr-only">Close Dialog</span>
       </button>
-      <h2 className="tw-mb-10 tw-text-3xl tw-leading-tight tw-font-semibold tw-text-center tw-normal-case">
-        {t(translations.buySovPage.slippageDialog.title)}
-      </h2>
 
-      <div className="tw-px-4">
-        <FormGroup label={t(translations.buySovPage.slippageDialog.tolerance)}>
-          <Slider
-            value={slippageValue}
-            onChange={e => setSlippageValue(e)}
-            min={0.1}
-            max={1}
-            stepSize={0.05}
-            labelRenderer={value => <>{value}%</>}
-            labelValues={[0.1, 0.25, 0.5, 0.75, 1]}
-            dataActionId={`${dataActionId}slippageDialog-slider`}
-          />
-        </FormGroup>
-        <br />
-        <FormGroup
-          label={t(translations.buySovPage.slippageDialog.minimumReceived)}
-        >
-          <div className={styles.dummy}>
-            <div data-action-id={`${dataActionId}slippageDialog-input`}>
-              <LoadableValue
-                value={<>{weiToNumberFormat(minReturn, 6)}</>}
-                loading={false}
-              />
-            </div>
-            <div>
-              <AssetRenderer asset={asset || Asset.SOV} />
-            </div>
-          </div>
-        </FormGroup>
+      <div className="tw-mw-340 tw-mx-auto">
+        <div className="tw-mb-6 text-left">
+          {t(translations.buySovPage.slippageDialog.title)}
+        </div>
+
+        <div className="tw-text-sm tw-font-light tw-tracking-normal">
+          <FormGroup
+            label={t(translations.buySovPage.slippageDialog.tolerance)}
+          >
+            <Slider
+              value={value}
+              onChange={handleChange}
+              min={0.1}
+              max={1}
+              stepSize={0.05}
+              labelRenderer={value => <>{value}%</>}
+              labelValues={[0.1, 0.25, 0.5, 0.75, 1]}
```

Comment:
This could be a constant outside the function.

Reply:
done